terraform/slurm_cluster/modules/slurm_controller_instance/main.tf

```diff
@@ -116,6 +116,41 @@ data "local_file" "cgroup_conf_tpl" {
   filename = local.cgroup_conf_tpl
 }
 
+###############################
+# DATA: PROLOG/EPILOG SCRIPTS #
+###############################
+
+data "local_file" "external_epilog" {
+  filename = "${path.module}/files/external_epilog.sh"
+}
+
+data "local_file" "external_prolog" {
+  filename = "${path.module}/files/external_prolog.sh"
+}
+
+data "local_file" "setup_external" {
+  filename = "${path.module}/files/setup_external.sh"
+}
+
+locals {
+  external_epilog = [{
+    filename = "z_external_epilog.sh"
+    content  = data.local_file.external_epilog.content
+  }]
+  external_prolog = [{
+    filename = "z_external_prolog.sh"
+    content  = data.local_file.external_prolog.content
+  }]
+  setup_external = [{
+    filename = "z_setup_external.sh"
+    content  = data.local_file.setup_external.content
+  }]
+
+  controller_startup_scripts = var.enable_external_prolog_epilog ? concat(local.setup_external, var.controller_startup_scripts) : var.controller_startup_scripts
+  epilog_scripts             = var.enable_external_prolog_epilog ? concat(local.external_epilog, var.epilog_scripts) : var.epilog_scripts
```

Comment:
Could you add a description of your testing to the pr description?
Have you tested running both with external and with provided var.epilog_scripts?

Reply:
Done, I thought I had included that.